### ISSUE TITLE
plot: ufunc 'multiply' did not contain a loop with signature matching types dtype('<U32') dtype('<U32') dtype('<U32')

### ISSUE THREAD

**Issue by rxin** [databricks/koalas#502]
I loaded the SF fire department dataset, and ... [fire-small.txt](https://github.com/databricks/koalas/files/3329631/fire-small.txt)

```
import databricks.koalas as ks
data = ks.read_csv("fire-small.txt", header=0, usecols=range(40))
data['Call Type'].plot.hist()
```

Error

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<command-3469248> in <module>()
----> 1 df['Call Type'].plot.hist()

/local_disk0/pythonVirtualEnvDirs/virtualEnv-b7eb6ad9-1ecb-4cf5-bb84-76d207603fc1/lib/python3.5/site-packages/databricks/koalas/plot.py in hist(self, bins, **kwds)
    540         axes : :class:`matplotlib.axes.Axes` or numpy.ndarray of them
    541         """
--> 542         return self(kind='hist', bins=bins, **kwds)
    543 
    544     def kde(self, bw_method=None, ind=None, **kwds):

/local_disk0/pythonVirtualEnvDirs/virtualEnv-b7eb6ad9-1ecb-4cf5-bb84-76d207603fc1/lib/python3.5/site-packages/databricks/koalas/plot.py in __call__(self, kind, ax, figsize, use_index, title, grid, legend, style, logx, logy, loglog, xticks, yticks, xlim, ylim, rot, fontsize, colormap, table, yerr, xerr, label, secondary_y, **kwds)
    466                            colormap=colormap, table=table, yerr=yerr,
    467                            xerr=xerr, label=label, secondary_y=secondary_y,
--> 468                            **kwds)
    469     __call__.__doc__ = plot_series.__doc__
    470 

/local_disk0/pythonVirtualEnvDirs/virtualEnv-b7eb6ad9-1ecb-4cf5-bb84-76d207603fc1/lib/python3.5/site-packages/databricks/koalas/plot.py in plot_series(data, kind, ax, figsize, use_index, title, grid, legend, style, logx, logy, loglog, xticks, yticks, xlim, ylim, rot, fontsize, colormap, table, yerr, xerr, label, secondary_y, **kwds)
    421                  yerr=yerr, xerr=xerr,
    422                  label=label, secondary_y=secondary_y,
--> 423                  **kwds)
    424 
    425 

/local_disk0/pythonVirtualEnvDirs/virtualEnv-b7eb6ad9-1ecb-4cf5-bb84-76d207603fc1/lib/python3.5/site-packages/databricks/koalas/plot.py in _plot(data, x, y, subplots, ax, kind, **kwds)
    437 
    438     plot_obj = klass(data, subplots=subplots, ax=ax, kind=kind, **kwds)
--> 439     plot_obj.generate()
    440     plot_obj.draw()
    441     return plot_obj.result

/local_disk0/pythonVirtualEnvDirs/virtualEnv-b7eb6ad9-1ecb-4cf5-bb84-76d207603fc1/lib/python3.5/site-packages/pandas/plotting/_core.py in generate(self)
    246 
    247     def generate(self):
--> 248         self._args_adjust()
    249         self._compute_plot_data()
    250         self._setup_subplots()

/local_disk0/pythonVirtualEnvDirs/virtualEnv-b7eb6ad9-1ecb-4cf5-bb84-76d207603fc1/lib/python3.5/site-packages/databricks/koalas/plot.py in _args_adjust(self)
    364             summary = KoalasHistPlotSummary(self.data, self.data.name)
    365             # computes boundaries for the column
--> 366             self.bins = summary.get_bins(self.bins)
    367 
    368         if is_list_like(self.bottom):

/local_disk0/pythonVirtualEnvDirs/virtualEnv-b7eb6ad9-1ecb-4cf5-bb84-76d207603fc1/lib/python3.5/site-packages/databricks/koalas/plot.py in get_bins(self, n_bins)
    330                       .collect()[0])
    331         # divides the boundaries into bins
--> 332         return np.linspace(boundaries[0], boundaries[1], n_bins + 1)
    333 
    334     def calc_histogram(self, bins):

/local_disk0/pythonVirtualEnvDirs/virtualEnv-b7eb6ad9-1ecb-4cf5-bb84-76d207603fc1/lib/python3.5/site-packages/numpy/core/function_base.py in linspace(start, stop, num, endpoint, retstep, dtype, axis)
    133     # Convert float/complex array scalars to float, gh-3504
    134     # and make sure one can use variables that have an __array_interface__, gh-6634
--> 135     start = asanyarray(start) * 1.0
    136     stop  = asanyarray(stop)  * 1.0
    137 

TypeError: ufunc 'multiply' did not contain a loop with signature matching types dtype('<U32') dtype('<U32') dtype('<U32')
```

**Comment by rxin** [comment]
cc @HyukjinKwon @dvgodoy

**Comment by HyukjinKwon** [comment]
@rxin, do you have the codes that created `df`? and it would be great if the data is uploaded here if that's small .. files can be uploaded into the comments here.

**Comment by rxin** [comment]
I've uploaded it.

**Comment by HyukjinKwon** [comment]
[fire-small.txt](https://github.com/databricks/koalas/files/3329631/fire-small.txt)

**Comment by HyukjinKwon** [comment]
@rxin, does this work in Pandas too?

```python
>>> import pandas as pd
>>> data = pd.read_csv("fire-small.txt", header=0, usecols=range(40))
>>> data['Call Type'].plot.hist()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/pandas/plotting/_core.py", line 2834, in hist
    return self(kind='hist', bins=bins, **kwds)
  File "/usr/local/lib/python3.7/site-packages/pandas/plotting/_core.py", line 2742, in __call__
    **kwds)
  File "/usr/local/lib/python3.7/site-packages/pandas/plotting/_core.py", line 1998, in plot_series
    **kwds)
  File "/usr/local/lib/python3.7/site-packages/pandas/plotting/_core.py", line 1801, in _plot
    plot_obj.generate()
  File "/usr/local/lib/python3.7/site-packages/pandas/plotting/_core.py", line 249, in generate
    self._compute_plot_data()
  File "/usr/local/lib/python3.7/site-packages/pandas/plotting/_core.py", line 367, in _compute_plot_data
    'plot'.format(numeric_data.__class__.__name__))
TypeError: Empty 'DataFrame': no numeric data to plot
>>> data['Call Type']
0     Medical Incident
1     Medical Incident
2     Medical Incident
3     Medical Incident
4    Traffic Collision
5    Traffic Collision
6    Traffic Collision
7     Medical Incident
8     Medical Incident
Name: Call Type, dtype: object
```

or is my data wrong? I uploaded into your PR description.

**Comment by HyukjinKwon** [comment]
If that's the case, let me fix the error message.

**Comment by dvgodoy** [comment]
I agree with you @HyukjinKwon , fixing this means raising the proper error.
Histograms are meant to be used with numerical columns only. For string columns, a barplot should do the trick.

**Comment by rxin** [comment]
@dvgodoy mind submitting a pull request to fix this?

**Comment by dvgodoy** [comment]
@rxin @HyukjinKwon 
Just submitted a PR to fix this :-)

**Comment by ueshin** [comment]
I'd close this since this is fixed at #766.
